pkg/controller/controller.go

```diff
@@ -671,9 +671,18 @@ func (p *csiProvisioner) getPVCSource(options controller.ProvisionOptions) (*csi
 		return nil, fmt.Errorf("error, new PVC request must be greater than or equal in size to the specified PVC data source, requested %v but source is %v", requestedSize, sourcePVC.Spec.Size())
 	}
 
+	sourcePV, err := p.client.CoreV1().PersistentVolumes().Get(sourcePVC.Spec.VolumeName, metav1.GetOptions{})
```

Comment:
The `PVC.Spec.VolumeName` field is "optional", would it be more reliable to just use the `Volume` field of the PVC directly?  Then you can do something like `volumeHandleToId` to set the `VolumeId` field below.

Reply:
do we have `volume` field in PVC spec? 

The approach I followed here is to get the PV name from PVC and then fetch volumeID from PV spec. do we have any other way of getting the volume name from the PVC?

Reply:
Actually you're correct here, Spec.VolumeName is the only handle; and the CO should set this when the claim is bound so this should be fine.  You might want to add a check that the Spec.VolumeName value is not nil before issuing the Get call for the PV though.